### PR TITLE
Sanitation on pkg/ingress/

### PR DESCRIPTION
**Pull request by chaosaffe**
**What this PR does / why we need it**:
General code cleanup based on some linter runs. Specifically focused on:
- Error checking and logging
- Removing dead code
- Deduplication

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 
N/A

**Special notes for your reviewer**:
None

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by chaosaffe** [comment]
/approve
This is just a test

File: pkg/ingress/controller/openstack/octavia.go [hunk 1]
```diff
@@ -498,48 +498,42 @@ func (os *OpenStack) CreatePolicyRules(lbID, listenerID, poolID, host, path stri
...
-		log.WithFields(log.Fields{"type": l7policies.TypeHostName, "host": host, "policyID": policy.ID, "listenerID": listenerID}).Info("policy rule created")
-	}
-
-	if path != "" {
-		log.WithFields(log.Fields{"type": l7policies.TypePath, "path": path, "policyID": policy.ID, "listenerID": listenerID}).Debug("creating policy rule")
-
-		// Create PATH type rule
-		_, err = l7policies.CreateRule(os.octavia, policy.ID, l7policies.CreateRuleOpts{
+		fields = log.Fields{"type": l7policies.TypePath, "host": host, "policyID": policy.ID, "listenerID": listenerID}
+	} else if path != "" {
```
Comment (lingxiankong) [inline pkg/ingress/controller/openstack/octavia.go]: the `host` and `path` could be specified at the same time, so should not use `else if` here.

File: pkg/ingress/controller/controller.go [hunk 2]
```diff
@@ -414,36 +414,35 @@ func (c *Controller) processItem(event Event) error {
...
-		} else {
-			c.recorder.Event(ing, apiv1.EventTypeNormal, "Deleted", fmt.Sprintf("Ingress %s", key))
+			return err
 		}
+
+		c.recorder.Event(ing, apiv1.EventTypeNormal, "Deleted", fmt.Sprintf("Ingress %s", key))
+		return nil
+	}
+
+	log.WithFields(log.Fields{"ingress": key}).Infof("ingress %sd, will update openstack resources", verb)
```
Comment (lingxiankong) [inline pkg/ingress/controller/controller.go]: I personally don't think it's a good idea for the refactoring in this patch, although there is some code duplication in the original implementation, it's more clear for the logic.

**Comment by lingxiankong** [comment]
/lgtm

Thanks!

**Pull request merged**